pkg/reconciler/revision/revision.go

```diff
@@ -63,9 +64,23 @@ type Reconciler struct {
 var _ revisionreconciler.Interface = (*Reconciler)(nil)
 
 func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) error {
-	// The image digest has already been resolved.
-	if rev.Status.ImageDigest != "" {
-		return nil
+	if rev.Status.ImageDigests == nil {
+		rev.Status.ImageDigests = make([]v1.ImageDigests, len(rev.Spec.Containers))
```

Comment:
```suggestion
		rev.Status.ImageDigests = make([]v1.ImageDigests, 0, len(rev.Spec.Containers))
```

You want to set the capacity and not the length of the array. Otherwise the if statement on line 73 would be true and return nil without actually resolving the digest.

Can we add a test case for that if condition